File: library/src/main/java/cn/quickits/aoide/encoder/AoideSoftEncoder.java
package cn.quickits.aoide.encoder;

public class AoideSoftEncoder {

    static {
        System.loadLibrary("aoide");
    }

    public static native void mp3EncodeInit(int inSamplerate, int inChannel, int outSamplerate, int outBitrate, int quality);

    public static native int mp3EncodeWrite(short[] bufferLeft, short[] bufferRight, int count, byte[] mp3buf);

    public native static int mp3EncodeFlush(byte[] mp3buf);

    public native static void mp3EncodeClose();

}

File: library/src/main/java/cn/quickits/aoide/encoder/AoideSoftEncoder.kt
package cn.quickits.aoide.encoder

object AoideSoftEncoder {

    init {
        System.loadLibrary("aoide")
    }

    external fun mp3EncodeInit(inSamplerate: Int, inChannel: Int, outSamplerate: Int, outBitrate: Int, quality: Int)

    external fun mp3EncodeWrite(bufferLeft: ShortArray, bufferRight: ShortArray, count: Int, mp3buf: ByteArray): Int

    external fun mp3EncodeFlush(mp3buf: ByteArray): Int

    external fun mp3EncodeClose()

}
